### DISCUSSION TITLE
Search debounce for select filters

### DISCUSSION THREAD

**Discussion by KuenzelIT** [filamentphp/filament#14804] [Ideas]
Hey everyone,

is there a possibility to change the **search debounce** of a **select filter**? There is a `searchDebounce()` method for Selects in Forms, but I cannot find one for select filters. If it does not exist, maybe it could be added?

Thanks!

**Comment by SimonMacIntyre** [comment]
I noticed that Select has a `debounce()`, but it does not seem to function in my testing. I'm using `->multiple()` and since it does not debounce it is not a super great user experience if you want to quickly click a few items.

**Comment by KuenzelIT** [comment]
I learned, that I have to modify the select form field like this:

`->modifyFormFieldUsing(fn($field) => $field->searchDebounce(CustomFilamentDefaults::$searchDebounce))`

This is working great. But I'm still looking for a solution to do set this generally on all select filters or even all search functions in filament.
